File: src/domain/message.rs
use chrono::{DateTime, Utc};
use uuid::Uuid;

pub struct Message {
    pub id: Uuid,
    pub role: String,
    pub content: String,
    pub chat_id: Uuid,
    pub timestamp: DateTime<Utc>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

File: internal/domain/models/entity/message.go
package entity

import (
	"time"

	"github.com/yelaco/ai-document-backend/internal/domain/models/types"
)

type Message struct {
	ID        types.MessageID
	ChatID    types.ChatID
	UserID    types.UserID
	Content   string
	Role      string
	Timestamp time.Time
	CreatedAt time.Time
	UpdatedAt time.Time
}
